packages/compartment-mapper/src/types/external.d.ts

```diff
@@ -0,0 +1,350 @@
+  computeSha512: HashFn;
+  computeSourceLocation: ComputeSourceLocationHook;
+  computeSourceMapLocation: ComputeSourceMapLocationHook;
+}> &
+  ModulesOption &
+  CompartmentOption &
+  ParserForLanguageOption &
+  LanguageForExtensionOption &
+  ExitModuleImportHookOption;
+
+export type LoadArchiveOptions = ParseArchiveOptions;
+
+type MapNodeModulesOptions = MapNodeModulesOptionsOmitPolicy & PolicyOption;
+type MapNodeModulesOptionsOmitPolicy = Partial<{
+  /** @deprecated renamed `conditions` to be consistent with Node.js */
+  tags: Set<string>;
+  conditions: Set<string>;
+  /**
+   * @deprecated Whether to make the devDependencies of the entry package
+   * accessible, implied by the `development` condition.
+   */
+  dev: boolean;
+  /** Dependencies to make reachable from any package */
+  commonDependencies: Record<string, string>;
+  /** Maps extensions to languages for all packages, like `txt` to `text` */
+  languageForExtension: LanguageForExtension;
+  /**
+   * Accounts for languages not present as values in any of the extension to
+   * language mappings.
+   * For higher level functions like `importLocation`, these are inferred
+   * from the `parses`.
+   */
+  languages: Array<Language>;
+}>;
+
+/**
+ * @deprecated Use `mapNodeModules()`.
+ */
+export type CompartmentMapForNodeModulesOptions = Omit<
+  MapNodeModulesOptions,
+  'conditions' | 'tags'
+>;
+
+export type CaptureLiteOptions = ImportingOptions &
+  LinkingOptions &
+  PolicyOption;
+
+export type ArchiveLiteOptions = SyncOrAsyncArchiveOptions &
+  ModuleTransformsOption &
+  ImportingOptions &
+  ExitModuleImportHookOption &
+  LinkingOptions;
+
+export type SyncArchiveLiteOptions = SyncOrAsyncArchiveOptions &
+  SyncModuleTransformsOption &
+  SyncImportingOptions &
+  ExitModuleImportNowHookOption;
+
+export type ArchiveOptions = Omit<MapNodeModulesOptions, 'language'> &
+  ArchiveLiteOptions;
+
+export type SyncArchiveOptions = Omit<MapNodeModulesOptions, 'languages'> &
+  SyncArchiveLiteOptions;
+
+/**
+ * Options for `loadLocation()`
+ */
+export type LoadLocationOptions = ArchiveOptions & SyncArchiveOptions;
+
+/**
+ * Options for `importLocation()` necessary (but not sufficient--see
+ * {@link ReadNowPowers}) for dynamic require support
+ */
+export type SyncImportLocationOptions = SyncArchiveOptions & ExecuteOptions;
+
+export type ImportLocationOptions = ArchiveOptions & ExecuteOptions;
+
+// ////////////////////////////////////////////////////////////////////////////////
+// Single Options
+
+type SearchSuffixesOption = {
+  /**
+   * Suffixes to search if the unmodified specifier is not found. Pass `[]` to
+   * emulate Node.js' strict behavior. The default handles Node.js' CommonJS
+   * behavior.
+   * Unlike Node.js, the Compartment Mapper lifts CommonJS up, more like a
+   * bundler, and does not attempt to vary the behavior of resolution depending
+   * on the language of the importing module.
+   */
+  searchSuffixes?: string[];
```

Comment:
`string[]` or `Array<string>`???

Reply:
I will change this to `Array` because that’s the prevailing convention, but really want to ignore this discrepancy unless there’s a meaningful difference and a lint rule to enforce it.